### PR TITLE
Precland fixes

### PR DESCRIPTION
**Pull request by okalachev**
This PR addresses two precland-related isses:

1. When the copter switches to the horizontal approach state, the check of landing target pose validity is not fully correct: an old message (when the landing target is seen the last time) can switch the state.

2. When downloading mission with a precision landing waypoint from the copter, the precision land parameter is lost.

@ndepal

### CODE REVIEW

File: src/modules/navigator/precland.cpp [hunk 1]
```diff
@@ -477,7 +477,7 @@ bool PrecLand::check_state_conditions(PrecLandState state)
 		}
 
 		// If we're trying to switch to this state, the target needs to be visible
-		return _target_pose_updated && _target_pose.abs_pos_valid;
+		return _target_pose_updated && _target_pose_valid && _target_pose.abs_pos_valid;
```
Comment (ndepal) [inline src/modules/navigator/precland.cpp]: Why is this necessary? `_target_pose_updated` is only true if we got a new message in this `on_active()` call. In this case, the timestamp of the message is recent, so `_target_pose_valid` is always true if `_target_pose_updated` is true.

It's only possible for `_target_pose_valid` to be true and `_target_pose_updated` false (due to `_param_timeout`), but not the other way around.

Comment (okalachev) [reply src/modules/navigator/precland.cpp]: Unfortunately, no. When testing we have met situations, when the copter sees the landing marker, when not in Precland mode (so Precland's `on_active` is not called). Then, for example, the copter flies from another point to the landing point. And then, "sees" the marker (actually just gets and old message).

Comment (ndepal) [reply src/modules/navigator/precland.cpp]: Ah I see. Thanks for the fix!
